src/Unused/Regex.hs

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Unused.Regex
+    ( matchRegex
+    ) where
+
+import Text.Regex.TDFA
+
+matchRegex :: String -> String -> Bool
+matchRegex r =
+    matchTest $ mkRegex r
+
+mkRegex :: RegexMaker Regex CompOption ExecOption String => String -> Regex
```

Comment:
Your original version makes more sense now that I've looked through the Regex types (everything is polymorphic!). Maybe `stringtoRegex` would be a good name to clarify what's happening?